pkg/client/query/cache/memory_test.go

```diff
@@ -0,0 +1,339 @@
+package cache
```

Comment:
@bryanchriswhite I'm not reviewing this but lmk if there's a particular section in the tests you'd want me to take a look at.